types/resource.ts

```diff
@@ -0,0 +1,12 @@
+export class Resource {
+  contributor?: string;
+  author?: string;
+  title?: string;
+  summary?: string;
+  source?: string;
+  level?: string;
+  mediaType?: string;
+  blockchain?: string[];
+  category?: string[];
+  tags?: string[];
+}
```

Comment:
Do you think it's worth it to make `contributor` and `source` mandatory?

Reply:
Ideally they would be mandatory, and I guess we could make sure that we have these details before starting the capture. The only reason any of these are optional is because we build up the structure over time.
If you can suggest a proper TS based pattern for this, some sort of builder, then we could do that.

Reply:
Yeah, I agree. The process that is in place right now will collect all of the required fields before reaching a point of printing out the structure. The only thing it doesn't do right now is the Author and Contributor as I haven't looked at the AirTable side of things at all.

Reply:
I've tried a slightly different approach to this, so that the Resource type has all fields required. I determined this type from looking at the Airtable form for adding a new Resource. (After double checking it I need to change one more thing)